pandas/core/index.py

```diff
@@ -1548,8 +1548,15 @@ def sym_diff(self, other, result_name=None):
         other, result_name_update = self._convert_can_do_setop(other)
         if result_name is None:
             result_name = result_name_update
-        the_diff = sorted(set((self.difference(other)).union(other.difference(self))))
-        return Index(the_diff, name=result_name)
+        the_diff = self.difference(other).union(other.difference(self))
+        the_diff_sorted = np.sort(the_diff.unique())
```

Comment:
I tried that, but the python `sorted` and `set` operations caused some errors with some of the types, and I'd imagine these would be better in numpy rather than python? 
But I can look into it again if you think worthy.